src/unxt/_src/units/system/core.py

```diff
@@ -213,6 +220,51 @@ def unitsystem(flag: type[StandardUnitSystemFlag], *units_: Any) -> AbstractUnit
     return unitsystem(*units_)
 
 
+@dispatch  # type: ignore[no-redef]
+def unitsystem(
+    flag: type[DynamicalSimulationUnitSystemFlag],
+    *units_: Any,
+    G: u.Quantity | float | int = 1.0,  # noqa: N803
+) -> AbstractUnitSystem:
+    tmp = unitsystem(*units_)
+
+    G = u.Quantity(G)
+
+    if G.unit.physical_type == u.get_physical_type("dimensionless"):
+        G = G * const_G
+    elif not G.unit.is_equivalent(const_G.unit):
+        msg = (
+            "Invalid value for G: must be dimensionless or have the correct dimensions "
+            "(equivalent to length per mass times velocity^2)"
+        )
+        raise ValueError(msg)
+
+    added = ()
+    if ud.length in tmp.base_dimensions and ud.mass in tmp.base_dimensions:
```

Comment:
This somehow feels like a match statement.

Reply:
Hm, I don't think there's a straightforward way to rewrite this with a match that is any clearer! If `tmp.base_dimensions` only contained length, mass, time then it could work, but since it can also contain other unit types (angle, energy, etc.), match cases don't support checking partial list matching AFAIK.